content/en/docs/contributing/building.md

```diff
@@ -0,0 +1,121 @@
+
+### Just update everything you can!
+
+Bazel takes care of a lot of automatic code generation for us, from generating CRD updates to updating it's own `BUILD.bazel` files.
+If you just want to do everything at once (and have 5 minutes of your time) you can run:
+
+```bash 
+$ ./hack/update-all.sh
+```
+
+This will update everything you need without having to care about what needs changing.
+
+> **NOTE:** we strongly reccomend to run this before you create a PR!
+
+### I need granular control 
+
+You can also pick and mix the individual bash helper scripts:
+* `update-bazel.sh`: updates the all `*.bazel` files includding formatting them
+* `update-codegen.sh`: runs all code generation
+* `update-crds.sh`: updates all CRD files to the lastest scheme
+* `update-deps-licenses.sh`: updates the `LICENCES` file, needed when adding/updating dependancies
+* `update-deps.sh`: installs new dependancies declared in the code and adds them into the Bazel and Go module files
+* `update-gofmt.sh`: runs `go fmt` on all code
+
+Most of these have a `verify-*` equivalent which will run inside our CI to verify all the scripts ran before merging the PR.
+
+## Building the project
+
+You can ask Bazel to build the code for you to run in your local machine.
+The output will end up in `bazel-out` on your disk.
+
+You can get the exact path by looking at the Bazel output:
+```
+Target //cmd/ctl:ctl up-to-date:
+  bazel-out/k8-fastbuild-ST-4c64f0b3d5c7/bin/cmd/ctl/kubectl-cert_manager
+```
+
+### Building the Go binaries (for local OS)
+
+You can build the controllers to run them locally using:
+```bash
+$ bazel build //cmd/...
+```
+If you need them inside a local cluster check out [our kind documentation](../kind/).
+
+### Building the CLI
+
+You can use `go run ./cmd/ctl` to quickly run the CLI.
+You can also compile it using Bazel:
+```bash
+$ bazel build //cmd/ctl
+```
+
+### Building images
+
+If you need the Docker images you can generate these using:
+```bash
+$ export APP_VERSION="dev"
+$ export DOCKER_REGISTRY="quay.io/jetstack/"
+$ bazel run \
+		--stamp \
+		--platforms=@io_bazel_rules_go//go/toolchain:linux_amd64 \
+		//build:server-images
+```
+
+`--stamp` enabled reproducable builds while `--platforms` defines which images to build, in this example for AMD64 Linux.
+
+## Testing the project 
+
+cert-manager got 3 kinds of tests, each one has their own way to invoke them to give you granular control
+
+* Unit tests: you can use `go test` (or your IDE) here, if Bazel using for example `bazel test //pkg/acme/...` for all tests in the ACME package
+* Integration tests: `bazel test //test/integration/...` this will run all integration tests against a Bazel operated `kube-apiserver`
+* End-to-end tests: see the [e2e documentation](../e2e/)
+
+> **TIP**: `...` is a recursive lookup in Bazel, it will run al tests in all subfolders when set, it is also the easiest way to invoke them.
```

Comment:
```suggestion
> **TIP**: `...` is a recursive lookup in Bazel, it will run all tests in all subfolders when set, it is also the easiest way to invoke them.
```